src/GLX/libglx.c

```diff
@@ -1275,13 +1275,13 @@ static char *MergeExtensionStrings(char *currentString, const char *newString)
     name = newString;
     nameLen = 0;
     while (FindNextExtensionName(&name, &nameLen)) {
-        if (!IsExtensionInString(currentString, name, nameLen)) {
+        if (!IsExtensionInString(buf, name, nameLen)) {
             *ptr++ = ' ';
             memcpy(ptr, name, nameLen);
             ptr += nameLen;
+            *ptr = '\0';
```

Comment:
It is perhaps not very clear. I have a patch that replaces the realloc with a malloc+memcpy and then frees currentString in the end. Let me know if you would prefer that.